Assignment/StackAndQueueImplementation/QueueImplementationUsingArray.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Queue implementation using array
+ * @author Rachna Jadam
+ * 
+ */
+public class QueueImplementationUsingArray<T> implements Queue<T> {
+    private Object[] queueArray;
+    private int front;
+    private int rear;
+    private int currentSize;
+    
+    public QueueImplementationUsingArray(int capacityOfQueue) {
+        queueArray = new Object[capacityOfQueue];
+        this.front = 0;
+        this.rear = 0;
+        this.currentSize = 0;
+    }
+    
+    /**
+     * this method adds element at the end (rear) of the queue.
+     * @param data  data that you want to insert in queue
+     */
+    @Override
+    public boolean enqueue(T data) throws AssertionError {
+    	if (data == null) {
+    	    throw new AssertionError("Data should not be null");
+    	}
+    	if (isQueueFull()) {
+            throw new AssertionError("Overflow ! Unable to add element in Queue");
+        }
+        if (rear++ >= queueArray.length && currentSize != queueArray.length) {
```

Comment:
Why are you using this condition??
According to me there is no need of this check if we already have checked that queue is full or not.